11_taxonomic_information.Rmd

```diff
@@ -175,6 +180,94 @@ altExp(tse, "Species_byPrevalence")
 assay(altExp(tse, "Species_byPrevalence"), "relabundance")[88:92, 1:7]
 ```
 
+```{r}
+# Saving the tse for later
+tseGlobalPatterns <- tse
+```
+
+### Taxa clustering {#taxa-clustering}
+Another way to agglomerate the data is to cluster the taxa. To do so,
+we usually start by doing a compositionality aware transformation such as CLR, 
+followed by the application of a standard clustering method.
+
+Here is an example that does a CLR transformation followed by the hierarchical
+clustering algorithm. 
+
+First, we import the library `bluster` that simplifies the clustering.
+```{r bluster_dependence}
+library(bluster)
+```
+
+Then we do the CLR transform followed by the clustering. We will cluster with
+two different distances: the euclidean distance and the kendall distance.
+
+```{r taxa_clustering}
+# Get the data
+data("peerj13075", package = "mia")
+tse <- peerj13075
+
+# The result of the CLR transform is stored in the assay clr
+tse <- transformAssay(tse, method = "clr", pseudocount = 1)
+
+tse <- transformAssay(tse, assay.type = "clr", method = "z", 
+                      MARGIN = "features")
+
+# Cluster (with euclidean distance) on the features of the z assay 
+tse <- cluster(tse, assay.type = "z",
+               clust.col = "hclustEuclidean", MARGIN = "features",
+               HclustParam(dist.fun = stats::dist, metric = "euclidean",
+                           method = "ward.D2"))
+
+# Declare the Kendall dissimilarity computation function
+kendall_dissimilarity <- function(x) {
+    as.dist(1 - cor(t(x), method = "kendall"))
+}
+
+# Cluster (with Kendall dissimilarity) on the features of the z assay 
+tse <- cluster(tse, assay.type = "z", MARGIN = "features", 
+               clust.col = "hclustKendall",
+               HclustParam(method = "ward.D2", 
+                           dist.fun = kendall_dissimilarity))
+```
+
+Let us store the resulting cluster indices in the `rowData` column specified 
+with the `clust.col` parameter.
+
+```{r taxa_clustering_result}
+# Checking the clusters
+clusters_euclidean <- rowData(tse)$hclustEuclidean
+head(clusters_euclidean, 10)
+
+clusters_kendall <- rowData(tse)$hclustKendall
+head(clusters_kendall, 10)
+```
+
+To better visualize the results and the distribution of the clusters, we can 
+plot the histogram of the clusters.
+
+```{r taxa_clustering_histogram}
+library(ggplot2)
+library(patchwork) # TO arrange several plots as a grid
+plot1 <- ggplot(as.data.frame(rowData(tse)), aes(x = clusters_euclidean)) + 
+    geom_bar() + 
+    labs(title = "CAG size distribution (Euclidean distance)",
+         x = "Clusters", y = "Feature count (n)")
+plot2 <- ggplot(as.data.frame(rowData(tse)), aes(x = clusters_kendall)) + 
+    geom_bar() + 
+    labs(title = "CAG size distribution (1 - tau distance)",
```

Comment:
"(1 - tau distance)" -> "(1 - tau dissimilarity)" or just "(1-tau)"  - I suggest the latter